tikv/gc.go

```diff
@@ -94,72 +95,115 @@ func (s *KVStore) resolveLocks(ctx context.Context, safePoint uint64, concurrenc
 	return nil
 }
 
+type BaseLockResolver struct {
+	store Storage
+	bo    *Backoffer
+}
+
+func NewBaseLockResolver(store Storage) *BaseLockResolver {
+	return &BaseLockResolver{
+		store: store,
+	}
+}
+
+func (l *BaseLockResolver) ResolveLocks(ctx context.Context, locks []*txnlock.Lock, loc *locate.KeyLocation) (*locate.KeyLocation, error) {
```

Comment:
* The name `BaseLockResolver` looks confused with those stuff in `lock_resovler.go`. Consider distinguish them by naming this `BaseGCLockResolver`. (and btw what does `Base` mean here?)
* The function `ResolveLocks` looks also confused with those methods in `lock_resolver.go`, which is used by transactions that meet stale locks. `BatchResolveLocks`, which is used for GC, has some differences with `ResolveLocks`. Here I suggest you to consider still using the name `BatchResolveLocks`, or more clearly, using somethig like `ResolveLocksForGC`.

Reply:
For TiDB GC Worker. it use BaseLockResolver to scan and resolve locks.
For TiDB Advacner(log backup). it use BaseLockResolver to scan locks and **override** `ResolveLocks` to handle locks with a small region range and check txn status one by one.(which is not BatchResolveLocks, force delete locks before safepoint)

So, BaseGCLockResolver is ok for me. and `BatchResolveLocks` seems not clear.